### PR TITLE
Image sharer task 5

### LINKED ISSUE #5: Add Style
#### As a user I want to see a visually pleasing application.

__Story__:

Our web application is really starting taking shape. We have added a few
features that provide value to our customers, but some of our customers are
complaining that the website is a quite bland. It is time to spice it up by
applying a visually appealing, and easy to maintain user interface to our web
application.

__Acceptance criteria__:
- [x] The application has a header with navigation
- [x] Images are displayed in an aesthetically pleasing list of cards
- [x] Form errors are shown inline with the related input field
- [x] I know how to use Bootstrap components like buttons and navbars in my
  application.
- [x] I know the very basics of CSS and SCSS
- [ ] My SCSS files are properly named and organized using BEM convention
- [x] My app has a "sticky" footer.
- [ ] My app has a few customized styles using bootstrap variable overrides.
- [x] My app is responsive and is usable on a phone.
- [x] The active link in the navigation header is differentiated.
- [x] I use a bootstrap "container" to center my app contents (not necessarily "fluid").

__Reference material__:
- http://v4-alpha.getbootstrap.com/getting-started/introduction/
- https://github.com/plataformatec/simple_form
- http://getbem.com/naming/

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by revatidamle**
Closes #5 

Trello card: https://trello.com/c/06UNCyQ8/1823-rails-on-boarding-task-5
---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/views/images/new.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<%= simple_form_for @image do |i| %>
+<%= simple_form_for @image, defaults: { wrapper_html: { class: 'form_group dfgdfg' }, input_html: { class: 'form-control' } } do |i| %>
```
Comment (joelbandi) [inline app/views/images/new.html.erb]: This class seems unintentional `dfgdfg`

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -4,6 +4,15 @@
     <th>Title</th>
     <th>Image</th>
   </tr>
+<nav class="navbar navbar-light bg-light">
+  <div style="text-align:center color: white">Image sharer app</div><br>
+  <form class="form-inline">
```
Comment (joelbandi) [inline app/views/images/index.html.erb]: Do you need to form for just one button? Seems like overkill

File: app/assets/stylesheets/_card.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.card-img-top {
```
Comment (joelbandi) [inline app/assets/stylesheets/_card.scss]: nitpicky: We can give it a better name like `.thumbnail`

File: app/assets/stylesheets/main.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  height: 100px;
+  bottom: 0;
+  width: 100%;
+  text-align: center;
+}
+
+#logo {
+  background-image: url("https://www.google.com/url?sa=i&source=images&cd=&ved=2ahUKEwjN1ObIz-HgAhWS_1QKHdD-BtMQjRx6BAgBEAU&url=https%3A%2F%2Fgithub.com%2Flogos&psig=AOvVaw3sOFNmb9tgh73zQ7UD0SN1&ust=1551553025208973");
+  display: block;
+  #margin: 0 auto;
```
Comment (joelbandi) [inline app/assets/stylesheets/main.scss]: remove all comments :)

File: app/views/images/_card.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<div class="card">
+  <img class="card-img-top" src="<%=image[:link]%>" />
+  <div class="card-body">
```
Comment (joelbandi) [inline app/views/images/_card.html.erb]: super nitpicky: indentation is missing here..

File: app/views/images/index.html.erb [hunk 6]
```diff
@@ -13,10 +7,16 @@
...
 
+<div class="container-fluid">
   <% @all_images.each do |image|%>
-      <tr id= <%= 'id_' + image[:title] %>>
-        <td><%=image.title %></td>
-        <td><%= image_tag  image.link, width: 400, crop: "pad"%></td>
-      </tr>
-  <%end %>
-</table>
+       <%= render :partial => 'images/card.html.erb', :locals => { :image => image } %>
```
Comment (joelbandi) [inline app/views/images/index.html.erb]: 👍

File: test/controllers/images_controller_test.rb [hunk 7]
```diff
@@ -23,32 +23,4 @@ def test_create__invalid_image
       assert_select '#image_link', 'ht://cdn.pixabay.com/photo/2013/08/22/19/18/rose-174817__340.jpg'
     end
   end
-
-  def test_index__check_order_of_display
```
Comment (joelbandi) [inline test/controllers/images_controller_test.rb]: Why did you delete this ?

Comment (revatidamle) [reply test/controllers/images_controller_test.rb]: This tested the table flow. We changed it to cards now. I could redo it for checking the card tag?

File: app/assets/stylesheets/main.scss [hunk 8]
```diff
@@ -0,0 +1,26 @@
+.footer-margin {
+  height: 100px;
+  bottom: 0;
+  width: 100%;
+  text-align: center;
+}
+
+#logo {
+  background-image: url("https://www.google.com/url?sa=i&source=images&cd=&ved=2ahUKEwjN1ObIz-HgAhWS_1QKHdD-BtMQjRx6BAgBEAU&url=https%3A%2F%2Fgithub.com%2Flogos&psig=AOvVaw3sOFNmb9tgh73zQ7UD0SN1&ust=1551553025208973");
```
Comment (grahamotte) [inline app/assets/stylesheets/main.scss]: Probably worth moving this asset to your `public` dir

File: app/views/images/index.html.erb (hunk 2 above)
Comment (grahamotte) [reply app/views/images/index.html.erb]: +1 this button is really just a link

File: app/views/images/index.html.erb [hunk 9]
```diff
@@ -1,13 +1,22 @@
...
+
+<div class="container-fluid">
   <% @all_images.each do |image|%>
-      <tr id= <%= 'id_' + image[:title] %>>
-        <td><%=image.title %></td>
-        <td><%= image_tag  image.link, width: 400, crop: "pad"%></td>
-      </tr>
-  <%end %>
-</table>
+       <%= render :partial => 'images/card.html.erb', :locals => { :image => image } %>
```
Comment (grahamotte) [inline app/views/images/index.html.erb]: Do we need to specify the `.html.erb` here?

File: test/controllers/images_controller_test.rb (hunk 7 above)
Comment (joelbandi) [reply test/controllers/images_controller_test.rb]: sounds good

File: app/views/images/index.html.erb (hunk 6 above)
Comment (joelbandi) [reply app/views/images/index.html.erb]: Also the new post ruby 1.9(i think) way of writing hashes is `symbol: 'value'` instead of `:symbol => 'value`

File: app/views/images/index.html.erb (hunk 9 above)
Comment (revatidamle) [reply app/views/images/index.html.erb]: Nope. Removed it. 👍

File: app/assets/stylesheets/application.scss [hunk 10]
```diff
@@ -11,5 +11,7 @@
  * It is generally better to create a new file per style scope.
  *
  *= require_tree .
- *= require_self
+
+ //*= require_self
```
Comment (grahamotte) [inline app/assets/stylesheets/application.scss]: Why add the `//` here?

File: app/assets/stylesheets/_card.scss [hunk 11]
```diff
@@ -0,0 +1,11 @@
+#thumbnail {
```
Comment (grahamotte) [inline app/assets/stylesheets/_card.scss]: Neither this id nor the class are great descriptors of what they're used for. We try to keep with BEM when possible http://getbem.com/introduction/ I don't think it's a big deal for this pr but would be worth using going forward

File: app/views/images/index.html.erb [hunk 12]
```diff
@@ -1,13 +1,20 @@
-<h4> Listing all images saved </h4>
-<table>
-  <tr>
-    <th>Title</th>
-    <th>Image</th>
-  </tr>
+<nav class="navbar navbar-light bg-light">
+  <div style="text-align:center color: white">Image sharer app</div><br>
+    <div style="padding-left: 10px;">
```
Comment (grahamotte) [inline app/views/images/index.html.erb]: Should be moved to a stylesheet

File: app/assets/stylesheets/application.scss (hunk 10 above)
Comment (revatidamle) [reply app/assets/stylesheets/application.scss]: I read in a blog for bootstrap installation that it is a good idea to remove the require_self from the application.scss file. Hence I commented it. Although either ways, it is not making a difference. Should I remove it?

File: app/assets/stylesheets/_card.scss (hunk 11 above)
Comment (revatidamle) [reply app/assets/stylesheets/_card.scss]: 👍 Will keep that in mid going further.

File: app/assets/stylesheets/application.scss (hunk 10 above)
Comment (grahamotte) [reply app/assets/stylesheets/application.scss]: I'd just remove it, better to not have commented out code
